### ISSUE TITLE
In 7.1.4, issues between `lerna version`, prettier 3, unhoisted prettier-plugin-svelte, and how to find the prettier ignore file when it is not in the default location

### ISSUE THREAD

**Issue by nstuyvesant** [lerna/lerna#3788]
## Current Behavior

Performed a `npx lerna version --yes --force-publish --conventional-commits --create-release github` and got errors (listed below)...

1. `lerna version` fails because `prettier@3.0.0` is in the root-level node_modules directory and `prettier-plugin-svelte@3.0.0` is in packages/svelte/node_modules (ie not hoisted).
2. `lerna version` assumes the default `.prettierignore`. In our case, `prettier` and `eslint` share the `.eslintignore` file at the root of the monorepo. There appears to be no way to pass in a prettier `--ignore-path` to `lerna version`.

Not sure I understand why `lerna version` needs prettier unless it was running pre-commit hooks. We have a pre-commit hook script that calls `lerna run format`. But the `format` script in each package's package.json works without hoisting `prettier-plugin-svelte`. Also, it includes the `--ignore-path`.

## Expected Behavior

`lerna version`, if it causes the pre-commit hooks will run `lerna run format` as well as it works from a command line. If there's another reason it uses prettier, I could not find docs on this.

## Steps to Reproduce

```bash
git clone https://github.com/carbon-design-system/carbon-charts.git
cd carbon-charts
yarn install
yarn build
yarn deploy
```

### Failure Logs / Configuration

```
lerna notice cli v7.1.4
lerna info ci enabled
lerna info current version 1.11.12
lerna WARN force-publish all packages
lerna info Assuming all packages changed
lerna info getChangelogConfig Successfully resolved preset "conventional-changelog-angular"

Changes:
 - @carbon/charts-angular: 1.11.12 => 1.11.13
 - @carbon/charts: 1.11.12 => 1.11.13
 - @carbon/charts-react: 1.11.12 => 1.11.13
 - @carbon/charts-svelte: 1.11.12 => 1.11.13
 - @carbon/charts-vue: 1.11.12 => 1.11.13

lerna info auto-confirmed 
lerna info execute Creating releases...
file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:19126
    Error.captureStackTrace(error);
          ^

Error: Cannot find package 'prettier-plugin-svelte' imported from /home/runner/work/carbon-charts/carbon-charts/noop.js
    at new NodeError (file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:19091:5)
    at packageResolve (file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:20036:9)
    at moduleResolve (file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:20068:20)
    at defaultResolve (file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:20173:15)
    at resolve (file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:20191:12)
    at importFromFile (file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:20206:15)
    at importFromDirectory (file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:20470:10)
    at file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:20487:12
    at async loadPlugin (file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:20492:28)
    at async Promise.all (index 0) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v18.16.1
```
After hoisting `prettier-plugin-svelte`...
```
file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:16967
        throw new UndefinedParserError(
              ^

UndefinedParserError: No parser could be inferred for file "/home/runner/work/carbon-charts/carbon-charts/yarn.lock".
    at normalizeFormatOptions (file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:16967:15)
    at formatWithCursor (file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:18588:52)
    at file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:21784:12
    at async Module.format2 (file:///home/runner/work/carbon-charts/carbon-charts/node_modules/prettier/index.mjs:21789:25)

Node.js v18.16.1
Error: Process completed with exit code 1.
```

<details><summary>lerna.json</summary><p>

```json
{
  "$schema": "node_modules/lerna/schemas/lerna-schema.json",
  "version": "1.11.12",
  "npmClient": "yarn",
  "command": {
    "publish": {
      "ignoreChanges": ["packages/*/demo/bundle", "**/*.md"]
    },
    "version": {
      "message": "chore(release): publish %v"
    }
  }
}
```

</p></details>

### Environment

lerna notice cli v7.1.4

 Environment info:

  System:
    OS: macOS 13.4.1
    CPU: (10) arm64 Apple M1 Pro
  Binaries:
    Node: 18.16.1 - /opt/homebrew/bin/node
    Yarn: 3.5.1 - /opt/homebrew/bin/yarn
    npm: 9.8.0 - /opt/homebrew/bin/npm
  Utilities:
    Git: 2.41.0 - /opt/homebrew/bin/git
  npmPackages:
    lerna: ^7.1.4 => 7.1.4
    prettier: ^3.0.0 => 3.0.0
    prettier-plugin-svelte: ^3.0.0 => 3.0.0

It took four failed releases (no way to test `lerna version` without running it) to successfully get a release out after upgrading the monorepo to prettier 3 - all trial and error.

To workaround the issues, I had to do two hacky things...
1. Add `prettier-plugin-svelte@3.0.0` to root-level package.json
2. Duplicate the contents of `.eslintignore` to a `.prettierignore` since lerna can't read our `.vscode/settings.json` and doesn't have a way for the ignore path to be passed in.

**Comment by nstuyvesant** [comment]
`prettier-plugin-svelte@3.0.0` specifically was for `prettier@3.0.0`

**Comment by nstuyvesant** [comment]
Probably important... `prettier-plugin-svelte` is NOT hoisted to the top level. It continues to live in the node_modules folder for the `svelte` package.

Not sure why `lerna version` even cares about prettier or prettier-plugin-svelte. Seems to me that formatting should be up to the author of the package.

**Comment by nstuyvesant** [comment]
My prettier ignore path is set in .vscode/settings.json. I am sharing the eslint ignore file.

**Comment by The-Code-Monkey** [comment]
Yeah I'm having a similar issue with the eslint-plugin-prettier latest version as they removed `resolvedPrettier.resolveConfig.sync` so this now breaks I assume it's the version of prettier and eslint in the background of lerna that's the issue because my linting runs fine on a prev step https://github.com/The-Code-Monkey/TechStack/actions/runs/5688521080/job/15418532358 here is the error in more detail

**Comment by elektronik2k5** [comment]
A simple possible way to avoid this problem is to add a flag (or configuration option) to not use prettier at all. Right now it is hardcoded. :(

**Comment by The-Code-Monkey** [comment]
I'm also getting this error too.

![Screenshot_20230730-141335](https://github.com/lerna/lerna/assets/31218527/51544ed6-5ef9-48cc-aee2-920847d00f02)

**Comment by nstuyvesant** [comment]
Does anyone know why `lerna version` uses prettier? This doesn't appear to be associated with my pre-commit hooks.

**Comment by The-Code-Monkey** [comment]
Yeah i can't understand why it does at all. I think it either needs
removing or a flag adding, also if it uses lerna it should be using the
most current version, or at least tested with compatibility

**Comment by The-Code-Monkey** [comment]
I might have a look at this on Tuesday if noone picks this up.

**Comment by fahslaj** [comment]
This was fixed by @BenoitZugmeyer ( #3796 ) and released in Lerna 7.1.5
